lib/modules/manager/poetry/schema.ts

```diff
@@ -99,7 +110,8 @@ const PoetryPypiDependency = z.union([
         managerData: { nestedVersion: true, sourceName: source?.toLowerCase() },
```

Comment:
Do not add sourceName if it is undefined.

Reply:
I've submitted a PR to fix this: #32162